### PR TITLE
Adding Config Reference & Development Docs plus Readme Makeover

### PR DESCRIPTION
**Pull request by louiseschmidtgen**
## Description
This PR is adding a reference page for our k8s-dqlite arguments, a development docs page and gives a general make-over to our existing pages.

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+This document describes the configuration options available for k8s-dqlite.
+
+## Configuration
+
+The following configuration options are available listed in a table format:
+
+| Option | Description | Default |
+|--------|-------------|---------|
+| `--storage-dir` | The directory to store the dqlite data | `/var/tmp/k8s-dqlite/` |
```
[the remark this answers is not in the document]
Comment (louiseschmidtgen) [reply docs/configuration.md:11]: Oh yes, that's a nice to have! I am not expecting these to change much to be honest.

File: README.md [hunk 2]
```diff
@@ -87,85 +88,34 @@ To point the API server to `k8s-dqlite` use the following arguments:
...
 
-Steps:
+### Steps
 
-1. On the main node, make sure that this node is listening to the default network interface.
-2. On the joining node and after installing k8s-dqlite, you need to backup the dqlite data directory for example (`/var/data`).
-3. Delete the dqlite data directory.
-4. Copy the `cluster.crt` and `cluster.key` from the main node and place it into the joining node's `/var/data`.
-5. Create the `init.yaml` file with the following content
+1. On the main node, ensure that `k8s-dqlite` is listening to the default network interface.
```
Comment (eaudetcobello) [inline README.md:96]: How does someone verify this?

File: docs/configuration.md (hunk 1 above)
Comment (eaudetcobello) [inline docs/configuration.md:11]: THANK YOU for this. Really nice.

File: docs/development.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+    ```bash
+    sudo snap install microk8s --classic
+    ```
+
+4. Wait for MicroK8s to come up:
+
+    ```bash
+    sudo microk8s status --wait-ready
+    ```
```
Comment (eaudetcobello) [inline docs/development.md:27]: Are we sure we want to recommend Microk8s instead of CK8s here?

**Review by eaudetcobello — COMMENTED** [review]
Nice work!

File: docs/development.md (hunk 3 above)
Comment (louiseschmidtgen) [reply docs/development.md:27]: I initially wanted to do that a few months back. The difference is listed a few lines further down. Now that this is moved into a separate dev doc out of the Readme we could make a second k8s-snap section.

File: README.md (hunk 2 above)
Comment (louiseschmidtgen) [reply README.md:96]: Great question!
It is not very clear but I assume this refers to the listen address. 
So for this example, we could do `sudo netstat -tuln | grep <the ip of the main node>:29001`

**Review by eaudetcobello — APPROVED** [review]

**Pull request merged**